merlin/spec/specification.py

```diff
@@ -354,8 +374,18 @@ def process_spec_defaults(self):
         if self.merlin["resources"]["workers"] is None:
             self.merlin["resources"]["workers"] = {"default_worker": defaults.WORKER}
         else:
-            for worker, vals in self.merlin["resources"]["workers"].items():
+            steps = self.get_study_step_names()
+            modified_steps = False
+            for _, vals in self.merlin["resources"]["workers"].items():
                 MerlinSpec.fill_missing_defaults(vals, defaults.WORKER)
+                for step in vals["steps"]:
```

Comment:
I’m finding this logic a little hard to follow. Maybe some comments could help?

Reply:
I just went through and added comments so you'll see them after I push changes. Essentially, this just goes through and makes sure every step in a workflow is assigned to a worker. I found a bug where if I didn't assign every step to a worker then the step that I didn't assign would never get completed. For example, say I had 2 steps `step_1` and `step_2` and 1 worker `demo_worker`. In the `steps` key for the worker I assigned it `steps: [step_1]`. This would run my workflow but would never run `step_2` since there was never a worker assigned to it. With the above change, `step_2` would be assigned to the default worker and would then get completed even without specifically assigning it to a worker.

Reply:
Ok! I see a little better now. Maybe I can suggest some variable name changes to help with confusion